### ISSUE TITLE
\dx+ fails due to missing pg_describe_object function

### ISSUE THREAD

**Issue by jmcatamney** [greenplum-db/gpdb#2574]
When I try to retrieve detailed information about installed extensions in a database, the \dx+ call errors out as follows:

```
testdb=# create extension plperl;
CREATE EXTENSION
testdb=# \dx
                List of installed extensions
  Name  | Version |   Schema   |         Description         
--------+---------+------------+-----------------------------
 plperl | 1.0     | pg_catalog | PL/Perl procedural language
(1 row)

testdb=# \dx+
ERROR:  function pg_catalog.pg_describe_object(oid, oid, integer) does not exist
LINE 1: SELECT pg_catalog.pg_describe_object(classid, objid, 0) AS "...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.
```

It looks like pg_describe_object was introduced in Postgres 9.1, so this is probably an issue of not backporting that function from upstream when extension support was backported.

**Comment by Eulerizeit** [comment]
```
testdb=# create extension plperl;
```
That syntax does not and will not work for a while. Currently GPDB has the extension framework but doesn't suppot and extensions. I don't see a reason to make this work until we have extensions to support.

**Comment by mkiyama** [comment]
I am confused.

The docs for the commands to manage extensions CREATE/ALTER/DROP EXTENSION
have been added to GPDB 5.0.

Should some other warning/note be added to the commands?

See Ivan's comment near the end of the PR.
https://github.com/greenplum-db/gpdb/pull/2363

**Comment by Eulerizeit** [comment]
Should we fix the psql that allows people to look for a feature we don't have yet? 

@ivannovick's point seemed to be that in the OSS version people can install extensions with some yak shaving. That is different from us having any extensions or us supporting the psql that shows the extensions we don't yet have.

**Comment by skahler-pivotal** [comment]
In the command line help we only show that \dx should work, we don't show that \dx+ as an option. I don't think we need to make this work, is there a way to make it fail more gracefully?

**Comment by Eulerizeit** [comment]
Now that I look at this, what I find slightly more disturbing is that 
```
testdb=# create extension plperl;
CREATE EXTENSION
testdb=# \dx
                List of installed extensions
  Name  | Version |   Schema   |         Description         
--------+---------+------------+-----------------------------
 plperl | 1.0     | pg_catalog | PL/Perl procedural language
(1 row)
```
Seems to have worked. Why is that?

**Comment by jmcatamney** [comment]
@skahler-pivotal I'm fine with just having \dx+ fail gracefully instead of fixing the query behind it; we don't need that functionality for anything at the moment, we just ran across the bug while doing some work with dumping extensions.

@Eulerizeit As per the linked PR, it looks like it's intended that CREATE EXTENSION works if the supporting files are already installed, but I'm not sure what happens if you try to create an extension without having the files present.

**Comment by hlinnaka** [comment]
1. In fact, `CREATE EXTENSION plperl` does work on GPDB 5. When we backported the extensions support, we did not convert all the contrib modules and languages to use the extensions support. But in commit 644b2a9d0b, we actually did that for pl/perl. It was quite possibly not intentional, at least I didn't explicitly mention it in the commit message. So it seems to work, but perhaps we should nevertheless remove those plperl extension files, for the sake of consistency.

I'm starting to feel that we should've bitten the bullet and converted all the contrib modules and languages into extensions, after all. The current situation, where we have extensions support but don't use it for anything that comes with the distribution, is a bit confusing.

2. If \dx+ doesn't work, that's a bug. We should fix it, one way or another. The trivial "fix" would be to just disable the command in psql. Or we could backport commit 6cc2deb86e, which added that function. That seems easy too, although it requires a catalog change.

**Comment by skahler-pivotal** [comment]
from a usage perspective, installing pl pieces with CREATE EXTENSION is much more intuitive IMHO and the presentation of the fact that they are installed is much cleaner. I don't see it as a big change in what we package for the tools in order to make that work. It is really just a couple more files and the largest amount of lines of something that need to be modified would be in docs. 

from a development perspective I think it also provides a much better framework to implement additional components

**Comment by Eulerizeit** [comment]
+ Lili since she owns the PL/* track

The question then becomes, can we change the way we install Languages and
MADlib and PostGIS without effecting the catalog?

-- Rob

**Comment by skahler-pivotal** [comment]
Anything in which we run a command to activate it in the database would need the treatment. This treatment essentially being gppkg still does the thing it does and then any further steps to activate it would be wrapped up in control file that would be executed when create extension is ran.

**Comment by ictmalili** [comment]
@skahler-pivotal Could you introduce more about your proposal for using "create extension" to install packages? For example, what level of packages you want to install, the whole package just as PL/Python, or a particular package such as NLTK?  Thanks

**Comment by hlinnaka** [comment]
This was fixed by PR #2979, closing.
